### PR TITLE
[phpunit 10] Add ParentTestClassConstructorRector

### PR DESCRIPTION
**Pull request by TomasVotruba**
Fixes BC break in case of non-test `TestCase` class use, ref https://github.com/sebastianbergmann/phpunit/commit/705874f1b867fd99865e43cb5eaea4e6d141582f

Ref https://github.com/sebastianbergmann/phpunit/issues/3975

### CODE REVIEW

File: rules-tests/PHPUnit100/Rector/Class_/ParentTestClassConstructorRector/Fixture/mocking_helper.php.inc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Rector\PHPUnit\Tests\PHPUnit100\Rector\Class_\ParentTestClassConstructorRector\Fixture;
+
+use PHPUnit\Framework\TestCase;
+
+final class MockingHelper extends TestCase
+{
+    public function __construct()
+    {
+        parent::__construct(static::class);
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply rules-tests/PHPUnit100/Rector/Class_/ParentTestClassConstructorRector/Fixture/mocking_helper.php.inc:23]: This should only happen in case of non-test and non-testcase. I'll update in a sec

**Comment by TomasVotruba** [comment]
Let's see how many edge-cases there will be in the wild :ship:

**Comment by samsonasik** [comment]
@TomasVotruba it seems constructor is marked as `final` since phpunit `12.0.3` so I think the rule `ParentTestClassConstructorRector` itself can be removed.

ref issue :

- https://github.com/rectorphp/rector/issues/9180

**Comment by samsonasik** [comment]
I created PR:

- https://github.com/rectorphp/rector-phpunit/pull/490

for it.

**Comment by samsonasik** [comment]
I created new PHPUnit 12 rule PR for it:

-  https://github.com/rectorphp/rector-phpunit/pull/491

to remove `__construct` on phpunit 12.
